docs/tutorials/ruby-driver-crud-operations.txt

```diff
@@ -458,20 +458,67 @@ the modification occurs.
   doc = artists.find_one_and_update( { :name => 'José James' }, { '$set' => { :name => 'José' } } )
   doc # Return the document before the update.
 
-``find_one_and_replace``
+Update Options
+~~~~~~~~~~~~~~
+
+To add options to an update command, specify them as key value pairs in the options
+Hash argument.
 
 .. code-block:: ruby
 
-  doc = artists.find(:name => 'José James').
-    find_one_and_replace( { '$set' => { :name => 'José' } }, :return_document => :after )
-  doc # Return the document after the update.
+  client = Mongo::Client.new([ '127.0.0.1:27017' ], :database => 'music')
+  artists = client[:artists]
 
-  doc = artists.find_one_and_replace(
-    { :name => 'José James' },
-    { '$set' => { :name => 'José' } },
-    :return_document => :after
+  # Use the _id index to perform this operation
+  result = artists.update_one(
+    { :name => 'Goldie' },
+    { "$inc" => { :plays =>  1 } },
+    { hint: { _id: 1 } }
   )
-  doc # Return the document after the update.
+  result.n # Returns 1.
+
+The following is a list of the options that can be added to update operations,
+including ``update_one``, ``update_many``, ``replace_one``,
+``find_one_and_delete``, ``find_one_and_update``, and ``find_one_and_replace``.
+
+.. list-table::
+   :header-rows: 1
+   :widths: 40 80
+
+   * - Option
+     - Description
+   * - ``array_filters``
+     - An Array of filter documents that determine which array elements to modify
+       for an update operation on an array field.
+   * - ``bypass_document_validation``
+     - Whether to skip document-level validation before writing the document.
+   * - ``collation``
+     - Specifies a set of rules to use when comparing strings complying with the
+       conventions of a particular language.
+   * - ``hint``
+     - The index to use for this operation. May be specified as a Hash
+       (e.g. { _id: 1 }) or as a String (e.g. "_id_").
+   * - ``projection``
+     - The fields to exclude or include in the operation result (only available
+       on ``find_one_and_delete``, ``find_one_and_replace``, and
+       ``find_one_and_update`` commands).
+   * - ``sort``
+     - How to sort the results of a findAndModify command. Specified as a Hash
+       key value pair, where the key is the name of the field to sort by, and
+       the value is either 1 or -1, specifying a sort in ascending or descending
+       order (only available on ``find_one_and_delete``, ``find_one_and_replace``,
+       and ``find_one_and_update`` commands).
+   * - ``session``
+     - The session to use for this operation.
+   * - ``upsert``
+     - Whether to upsert if the document doesn't exist. Cannot be used on
+       ``find_one_and_delete`` operation.
```

Comment:
What about the `:return_document` that was in the example previously, does it still work?

Reply:
Yup! Thanks for pointing that out.